normalize_license_text/normalize_class.py

```diff
@@ -0,0 +1,105 @@
+        normalized_string = self.lowercase()
+        normalized_string = self.equivalentwords(normalized_string)
+        normalized_string = self.copyrightsymbol(normalized_string)
+        normalized_string = self.bullets_numbering(normalized_string)
+        normalized_string = self.punctuation(normalized_string)
+        normalized_string = self.license_title(normalized_string)
+        normalized_string = self.remove_whitespace(normalized_string)
+        return normalized_string
+        
+    def lowercase(self):
+        normalized_string = self.inputstring.lower()
+        return normalized_string
+        
+    def equivalentwords(self,normalized_string):
+        try:
+            equivalentfile = open("equivalentwords.txt","r")
+            for word in equivalentfile.readlines():
+                splitwords = word.split(',')
+                wordtoreplace = splitwords[0]
+                wordreplaced = splitwords[1]
+                wordtoreplace = str(wordtoreplace)
+                wordreplaced = str(wordreplaced)[:-1]
+                normalized_string = re.sub(wordreplaced,wordtoreplace,normalized_string)
+            equivalentfile.close()
+            return normalized_string
+        except IOError:
+            print("There was some problem with the function.")
+        
+        
+    def copyrightsymbol(self,normalized_string):
+        
+        """ Removes the copyright symbol and the possibilities of mismatch due to it. """
+        try:
+            normalized_string = normalized_string.replace("copyright",'')
+            normalized_string = normalized_string.replace('(c)','')
+            return normalized_string
+        except IOError:
+            print("This function could not run properly.")
+                    
+    def bullets_numbering(self,normalized_string):
+        
+        """ The most important and error prone function is the handling of the bullets
+        and numbering cases. The case has just 1 error prone possibility and requirement
+        that the numbering should carry a space after it or else it gets matched with the 
+        version. 
+        (1. ) will match while (1.) will not.
+        """
+        try:
+            normalized_string = re.sub(r'([0-9]+\.){2,}','',normalized_string)
+            normalized_string = re.sub(r'[0-9]+\.[\D]','',normalized_string)
+            normalized_string = re.sub(r'^[a-z]\.','',normalized_string)
+            normalized_string = re.sub(r'[a-z]\)','',normalized_string)
+            normalized_string = re.sub(r'[0-9]\)','',normalized_string)
+            normalized_string = re.sub(r'^[A-Z]\.','',normalized_string)
+            normalized_string = re.sub(r'^[mdclxvi]+\.','',normalized_string)
+            return normalized_string
+        except IOError:
+            print("This function could not run properly.")
+
+    def punctuation(self,normalized_string):
+        
+        try:
+            normalized_string = normalized_string.replace('-','*')
```

Comment:
Keep all the characters that need to be replaces in a list and call a for loop on that list

Reply:
This will trim down the code alot